### PR TITLE
chore(ci): migrate tests to jest and add jenkins

### PR DESCRIPTION
**Pull request by ghmeier**
#### Changes Made
Adds a jenkins file, removes the travis file, and uses jest for testing.

Overall this conforms to a more widely supported test framework, removes
some ava-related oddities and uses a more efficient CI tool.

#### Potential Risks
small. this doesn't update any in-use code. the biggest risk is a regression in test coverage.

#### Test Plan
ensure tests pass

#### Checklist
- [x] I've increased test coverage
- [x] Since this is a public repository, I've checked I'm not publishing private data in the code, commit comments, or this PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,14 +44,15 @@
     "underscore": "^1.9.2"
   },
   "devDependencies": {
+    "@mixmaxhq/jest-coverage-config": "^2.0.1",
+    "@mixmaxhq/jest-junit-config": "^1.0.0",
```
Comment (skeggse) [inline package.json:48]: These are currently private, and I imagine we don't want to block external folks from `npm i`/`npm ci`.

File: spec/aggregateSpec.js [hunk 2]
```diff
@@ -1,418 +1,416 @@
...
+describe('aggregate', () => {
+  const t = {};
+  beforeAll(async () => {
+    mongod = dbUtils.start();
+    t.db = await dbUtils.db(mongod, driver);
+
+    // Set up collections once for testing later.
+    await Promise.all([
+      t.db.collection('test_paging').insert([
+        {
```
Comment (janecakemaster) [inline spec/aggregateSpec.js:18]: so
many
newlines
😭

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  testMatch: ['**/spec/**/*Spec.js'],
```
Comment (sbalaji1996) [inline jest.config.js:2]: Do you plan on migrating from spec -> test? Thought I saw a thread in slack about this

File: spec/support/db.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ const MongoClient = require('mongodb');
 
 function start() {
   return new MongoMemoryServer({
-    binary: { version: '3.4.6' },
+    binary: { version: '3.6.13' },
```
Comment (sbalaji1996) [inline spec/support/db.js:7]: It'd be cool if we could run tests against multiple different mongo versions (I know mongoist does this)

Comment (ghmeier) [reply spec/support/db.js:7]: True! That'd be nice.

File: jest.config.js (hunk 3 above)
Comment (ghmeier) [reply jest.config.js]: You got it!

**Pull request merged**